### ISSUE TITLE
Linter does not complain about missing @required values in parent class constructor

### ISSUE THREAD

**Issue by dotdoom** [dart-lang/linter#1708]
**Describe the issue**

When class A has a `@required` parameter in constructor, and class B does not invoke `super()`, the linter should complain about it.

**To Reproduce**

```dart
// required_in_parent_constructor.dart 
import 'package:meta/meta.dart';

class A {
  int value;
  A({@required this.value}) : assert(value != null);
}

class B extends A {}

void main() {
  print(B().value);
}
```

```shell
$ dartanalyzer required_in_parent_constructor.dart 
Analyzing required_in_parent_constructor.dart...
No issues found!

$ dart --enable-asserts required_in_parent_constructor.dart 
Unhandled exception:
<...>
'value != null': is not true.
```

**Expected behavior**

`dartanalyzer` complains about missing value for `@required` parameter.

**Comment by pq** [comment]
Good catch.  You're proposing we should flag `B` for not defining a constructor that calls `super(value: ...)` ?  Sounds sensible to me.

/cc @bwilkerson

**Comment by bwilkerson** [comment]
Yes, we should catch that case. This check is actually a hint in analyzer, though, not a lint. And I think there's an issue for it already in the sdk repo, though I didn't take the time to confirm that.

**Comment by pq** [comment]
@dotdoom: it looks like this one is being tracked already in https://github.com/dart-lang/sdk/issues/37266.  Feel to chime in there's anything you want to add.

Thanks for the report!

**Comment by dotdoom** [comment]
@pq thanks for redirecting it. I guess I messed it up again just as in https://github.com/dart-lang/sdk/issues/35577 -> dart-lang/linter#1395 -- filing an issue in the wrong repo.

**Comment by pq** [comment]
No worries at all @dotdoom.  It's not easy to find duplicates (even for us) and often not at all obvious where to file issues.  The most important thing is to keep getting feedback, so thanks again for doing that!
